# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BaselineBailouts.cpp
# Commit: 6ccaefced3df
# Full Hash: 6ccaefced3df63b297057b31a856527153801a71
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2020-01-13 15:45:24
# Description:
#   Bug 1605641 - Mark 'done' stack value for destructuring assignment as having implicit uses. r=tcampbell
#   
#   We need to make sure this value isn't optimized out because the exception
#   handler may need it.
#   
# ==============================================================================

diff -r 96753184f3c4 -r 6ccaefced3df js/src/jit/BaselineBailouts.cpp
--- a/js/src/jit/BaselineBailouts.cpp	Thu Jan 02 12:07:51 2020 +0000
+++ b/js/src/jit/BaselineBailouts.cpp	Mon Jan 13 09:23:17 2020 +0000
@@ -488,18 +488,6 @@
   return pc;
 }
 
-class NoOpTryNoteFilter {
- public:
-  explicit NoOpTryNoteFilter() = default;
-  bool operator()(const JSTryNote*) { return true; }
-};
-
-class TryNoteIterAll : public TryNoteIter<NoOpTryNoteFilter> {
- public:
-  TryNoteIterAll(JSContext* cx, JSScript* script, jsbytecode* pc)
-      : TryNoteIter(cx, script, pc, NoOpTryNoteFilter()) {}
-};
-
 static bool HasLiveStackValueAtDepth(JSContext* cx, HandleScript script,
                                      jsbytecode* pc, uint32_t stackSlotIndex,
                                      uint32_t stackDepth) {